src/ng-pattern-restrict.js

```diff
@@ -140,6 +140,12 @@ angular.module('ngPatternRestrict', [])
           function updateCurrentValue(newValue) {
             oldValue = newValue;
             caretPosition = getCaretPosition();
+            
+            if (ngModelController) {
```

Comment:
I just realized that these changes were added to the `updateCurrentValue` function, which only stores the value to be used as a valid replacement next time. At this point, should there be any call to update the view value, since it does not perform any change to the value?

As for the replacement values, you can see how this is done in the function `revertToPreviousValue`.